src/migrations/data.ts

```diff
@@ -383,6 +384,44 @@ function dropCommands(initialMessages, commandType: string) {
   return messages;
 }
 
+function fixChartDefinitions(data: Partial<WorkbookData>, initialMessages: StateUpdateMessage[]) {
+  const messages: StateUpdateMessage[] = [];
+  const map = {};
+  for (const sheet of data.sheets || []) {
+    sheet.figures?.forEach((figure) => {
+      if (figure.tag === "chart") {
+        // chart definition
+        map[figure.id] = figure.data;
+      }
+    });
+  }
+  for (const message of initialMessages) {
+    if (message.type === "REMOTE_REVISION") {
+      const commands: CoreCommand[] = [];
+      for (const cmd of message.commands) {
+        let command = cmd;
+        switch (cmd.type) {
+          case "CREATE_CHART":
+            map[cmd.id] = cmd.definition;
+            break;
+          case "UPDATE_CHART":
+            const definition = map[cmd.id];
+            command = { ...cmd, definition: { ...definition, ...cmd.definition } };
```

Comment:
I think you should add `map[cmd.id] = { ...definition, ...cmd.definition }`.

Because if you have two commands `UPDATE_CHART` which update different "fields", only the last one will be update the definition.

Let's say:
`CREATE_CHART({ name: "hello", dataSets: "A1" })` 
`UPDATE_CHART({ name: "bla" })`
`UPDATE_CHART({ dataSets: "A2" })`
The chart definition will be `{ name: "hello", dataSets: "A2" }`
Right ?